helper.js

```diff
@@ -115,12 +125,12 @@ exports.txInToString = (dir, txInList, isCollateral) => {
           ? `--tx-in-script-file ${this.jsonToPath(dir, txIn.script)} `
           : ""
       } ${
-        txIn.datum
-          ? `--tx-in-datum-value '${JSON.stringify(txIn.datum)}' `
+        txIn.datum || txIn.datumJSON
+          ? this.datumToString(dir, txIn.datum, txIn.datumJSON)
```

Comment:
IMHO calling directly `datumToString` and passing `txIn.datum` and `txIn.datumJSON` would remove some complexity. The function can handle all that logic and directly return a string.
The same applies to the redeemer part.